rust/flatbuffers/src/vector.rs

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
```

Comment:
Wow, not being a rustacean, all this follow/phantom/slice slice stuff is pretty mind-boggling. Doesn't need to be part of this PR, but would be good to at some point explain this stuff somewhere, not just for interested readers, but also for users that end up running into implementation details. Or to sell the specific benefit of what Rust + FlatBuffers allows, which may be beyond other languages.

Reply:
This is a great idea.